src/librustc/mir/mod.rs

```diff
@@ -395,6 +411,10 @@ pub struct LocalDecl<'tcx> {
     /// True if this corresponds to a user-declared local variable.
     pub is_user_variable: bool,
 
+    /// True if this an internal local.
+    /// Such locals are not checked against the legal types in a generator.
+    pub internal: bool,
```

Comment:
I'm not sure what is the best way to write that comment, but the idea (that the point of this is to disable sanity checks, and this is sound exactly for "fully algebraic" variables) should be expressed.